### DISCUSSION TITLE
Proposal: make `raise_for_status` chainable

### DISCUSSION THREAD

**Discussion by trim21** [encode/httpx#2775] [Ideas] [answered]
make it possible to write code like `r = httpx.get('...').raise_for_status().json()`

**Comment by karpetrosyan** [comment]
Hey! I moved your issue to discussion because that is where ideas like this start, and if it is approved, we can move forward by opening an issue.

**Reply by lovelydinosaur** [reply, marked as answer]
See pull request #2776.

(Reminder to all... let's start with escalating to an issue *if* a discussion is accepted, rather than jumping straight to a PR.)

Anyways... great thanks!
